pyvista/utilities/algorithms.py

```diff
@@ -0,0 +1,271 @@
+
+    port : int, default: 0
+        The input port
+
+    """
+    if isinstance(inp, _vtk.vtkAlgorithm):
+        alg.SetInputConnection(port, inp.GetOutputPort())
+    elif isinstance(inp, _vtk.vtkAlgorithmOutput):
+        alg.SetInputConnection(port, inp)
+    else:
+        alg.SetInputDataObject(port, inp)
+
+
+class PreserveTypeAlgorithmBase(_vtk.VTKPythonAlgorithmBase):
+    """Base algorithm to preserve type."""
+
+    def __init__(self, nInputPorts=1, nOutputPorts=1):
+        """Initialize algorithm."""
+        _vtk.VTKPythonAlgorithmBase.__init__(
+            self,
+            nInputPorts=nInputPorts,
+            nOutputPorts=nOutputPorts,
+        )
+
+    def GetInputData(self, inInfo, port, idx):
+        """Get input data object.
+
+        This will convert ``vtkPointSet`` to ``vtkPolyData``
+        """
+        inp = wrap(_vtk.VTKPythonAlgorithmBase.GetInputData(self, inInfo, port, idx))
+        if isinstance(inp, pyvista.PointSet):
+            return inp.cast_to_polydata()
+        return inp
+
+    # THIS IS CRUCIAL to preserve data type through filter
+    def RequestDataObject(self, request, inInfo, outInfo):
+        """Preserve data type."""
+        class_name = self.GetInputData(inInfo, 0, 0).GetClassName()
+        if class_name == 'vtkPointSet':
+            # See https://gitlab.kitware.com/vtk/vtk/-/issues/18771
+            self.OutputType = 'vtkPolyData'
+        else:
+            self.OutputType = class_name
+        self.FillOutputPortInformation(0, outInfo.GetInformationObject(0))
+        return 1
+
+
+class ActiveScalarsAlgorithm(PreserveTypeAlgorithmBase):
+    """Internal helper algorithm to control active scalars.
+
+    Parameters
+    ----------
+    name : str
+        Name of scalars used to "color" the mesh.  Accepts a
+        string name of an array that is present on the mesh.
+        Array should be sized as a single vector.
+
+    preference : str, optional
+        When ``mesh.n_points == mesh.n_cells`` and setting
+        scalars, this parameter sets how the scalars will be
+        mapped to the mesh.  Default ``'point'``, causes the
+        scalars will be associated with the mesh points.  Can be
+        either ``'point'`` or ``'cell'``.
+
+    """
+
+    def __init__(self, name, preference='point'):
+        """Initialize algorithm."""
+        super().__init__()
+        self.scalars_name = name
+        self.preference = preference
+
+    def RequestData(self, request, inInfo, outInfo):
+        """Perform algorithm execution."""
+        try:
+            inp = wrap(self.GetInputData(inInfo, 0, 0))
+            out = self.GetOutputData(outInfo, 0)
+            output = inp.copy()
+            if output.n_arrays:
+                output.set_active_scalars(self.scalars_name, preference=self.preference)
+            out.ShallowCopy(output)
+        except Exception as e:  # pragma: no cover
+            traceback.print_exc()
+            raise e
+        return 1
```

Comment:
Yes, this is a VTK-ism... effectively returning `True` to let the pipeline know that `RequestData` was completed successfully. Failing to do this will error out. Actually, in the block above where I have `traceback.print_exc()`, it should `return 0` to indicate failure.

I'll explain a bit further in the thread above on `try`-`except` soon